packages/common/src/search/Collection.ts

```diff
@@ -1,36 +1,68 @@
-import { hubSearchQuery, IArcGISContext } from "..";
+import { cloneObject, IArcGISContext } from "..";
 import {
   EntityType,
-  IHubCatalog,
   IHubCollection,
   IHubSearchOptions,
   IHubSearchResponse,
   IHubSearchResult,
   IQuery,
 } from "./types";
-
+import { hubSearch } from "./hubSearch";
 /**
- * Implements Collection behavior
+ * Collection Class
+ *
+ * Abstracts searching a Collection
+ *
+ * For more information, check out the [Catalog & Collection Guide](/hub.js/guides/concepts/catalog-collection/)
  */
-/* istanbul ignore next */
 export class Collection implements IHubCollection {
```

Comment:
Are we intending the `key` property to be an identifier unique among the collections of a resource's catalog? If so we might want to include that logic when trying to create a new Collection

Reply:
Within the context of a catalog, the key should be unique. But, if you're just creating a collection to pass into a Gallery, separate from a Catalog, the prop  just needs to be present. Agreed that when we add a `addCollection(...)` fn to the Catalog we need to implement this check, but since Collections can exist separate from Catalogs, we can't put that logic into the Collection class itself